### DISCUSSION TITLE
How to add approval permissions in Elsa 3.1.3 designer?

### DISCUSSION THREAD

**Discussion by shxylgithub** [elsa-workflows/elsa-core#5721] [General]
How to add approval permissions to nodes in Elsa 3.1.3 designer? For example, the first level approval is for the team leader, the second level approval is for the department manager, and the third level approval is for the general manager.
![approval](https://github.com/elsa-workflows/elsa-core/assets/171897414/bde9ae4a-062e-444d-b5cc-3c7adf67b9fc)

**Comment by glime-ai** [comment]
To add approval permissions to nodes in Elsa 3.1.3 designer, you would typically use a combination of user roles and activities that enforce these roles at each step of the workflow. Here's a general approach to achieve this:

1. **Define User Roles**: Define roles in your system such as "Team Leader", "Department Manager", and "General Manager". These roles should be associated with the user accounts in your system.

2. **Use Activities to Enforce Roles**: In your workflow, use activities that check the role of the user before allowing the workflow to proceed. This could be a custom activity that you create, which checks the current user's role and compares it with the required role for that step.

3. **Configure Approval Steps**: For each approval step in your workflow, configure the activity to require the appropriate role. For example, for the first level approval, you might have an activity that only allows users with the "Team Leader" role to approve.

4. **Handle Approvals**: Use activities such as "Send Email" to notify the appropriate approvers when their action is required. You can also use "Timer" activities to send reminders if the approval is pending for too long.

5. **Visual Representation**: If you want to visualize these approval steps with permissions in the Elsa Designer, you would represent each approval step as a node and configure the node with the appropriate role-checking logic.

Unfortunately, as of my knowledge cutoff date in April 2023, there isn't a direct feature in Elsa Designer that allows you to visually set permissions on nodes out-of-the-box. You would need to implement the logic within the activities that make up the nodes.

Please note that the link to the image you provided seems to be incorrect or incomplete, as it leads to a 404 page on GitHub. If you need further assistance with the specific implementation or if there are updates to Elsa after my knowledge cutoff date, I recommend checking the official Elsa documentation or reaching out to the Elsa community for the latest information.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
